### PR TITLE
Task/193 swap form UI

### PR DESCRIPTION
**Pull request by jvrastil**
- new exchange coins UI added (go to coin detail -> exchange)
- restyling sending coins UI according to the design

### CODE REVIEW

File: lib/view/widgets/toggle.dart [hunk 1]
```diff
@@ -23,44 +22,60 @@ class Toggle extends StatefulWidget {
 class _Toggle extends State<Toggle> {
   @override
   Widget build(BuildContext context) {
-    var borderRadius = BorderRadiuses.radius64;
+    final borderRadius = BorderRadii.radius64;
 
     return Container(
+      height: 28,
```
Comment (TomasPetrik) [inline lib/view/widgets/toggle.dart:28]: This could be defined as public constants from Constants class.

File: pubspec.yaml [hunk 2]
```diff
@@ -8,9 +8,8 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 # For production version we use tags on master branch, e.g. 'v1.2.4'
 version: 0.0.0+88
 
-
 environment:
-  sdk: ">=2.17.1 <3.0.0"
+  sdk: '>=2.17.1 <3.0.0'
```
Comment (TomasPetrik) [inline pubspec.yaml:12]: It's a good idea to replace all strings in our with single quotes:

https://dart-lang.github.io/linter/lints/prefer_single_quotes.html

Consider to make this job in separated task.

Comment (ciripel) [reply pubspec.yaml:12]: I agree with this

File: lib/logic/cubit/asset_exchange_form/asset_exchange_form_state.dart [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+          totalAmount: '0',
+          amountUnit: AmountUnit.crypto,
+          totalSwapFee: '0',
+          sourceTransactionFee: '0',
+          targetTransactionFee: '0',
+          swapFee: '0',
+        );
+
+  bool get isValid =>
+      // assetId >= 0 &&
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/logic/cubit/asset_exchange_form/asset_exchange_form_state.dart:73]: done

File: lib/logic/cubit/asset_send_form/asset_send_form_cubit.dart [hunk 4]
```diff
@@ -1,34 +1,67 @@
 import 'package:equatable/equatable.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:simplio_app/data/model/network_wallet.dart';
+import 'package:simplio_app/data/repositories/fee_repository.dart';
+import 'package:simplio_app/data/repositories/wallet_repository.dart';
+// TODO: Do we move this extension somewhere more general?
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/logic/cubit/asset_send_form/asset_send_form_cubit.dart:7]: removed

File: lib/logic/cubit/asset_send_form/asset_send_form_cubit.dart [hunk 5]
```diff
@@ -60,29 +100,184 @@ class AssetSendFormCubit extends Cubit<AssetSendFormState> {
...
-  void minAmountClicked() async {
-    debugPrint('minAmountClicked - NEEDS TO BE IMPLEMENTED');
+  void addDecimalDotFiat() async {
+    if (state.amountFiat.contains('.')) return;
+
+    var result = double.tryParse(state.amountFiat);
+    result ??= 0;
+
+    emit(state.copyWith(
+      amountFiat: result.toString().substring(
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/logic/cubit/asset_send_form/asset_send_form_cubit.dart:134]: 👍 done

File: pubspec.yaml (hunk 2 above)
Comment (ciripel) [reply pubspec.yaml:12]: but this is changed already 2 weeks ago! from where did u get this?

File: lib/data/repositories/wallet_repository.dart [hunk 6]
```diff
@@ -56,6 +56,10 @@ class WalletRepository {
     return w;
   }
 
+  String get getCurrentAccountWalletId {
+    return _walletId;
+  }
+
   Future<AccountWallet> addAccountWallet(
```
[the remark this answers is not in the document]
Comment (ciripel) [reply lib/data/repositories/wallet_repository.dart:63]: This method is not needed ... i will delete it!

File: lib/data/repositories/wallet_repository.dart [hunk 7]
```diff
@@ -56,6 +56,10 @@ class WalletRepository {
     return w;
   }
 
+  String get getCurrentAccountWalletId {
```
[the remark this answers is not in the document]
Comment (ciripel) [reply lib/data/repositories/wallet_repository.dart:59]: we will use them in the future

File: lib/view/helpers/custom_painters.dart [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    required this.color,
+    required this.start,
+    required this.end,
+  });
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final paint = Paint();
+    paint.style = PaintingStyle.stroke;
+    paint.strokeWidth = 2.0;
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/helpers/custom_painters.dart:18]: done

File: lib/view/helpers/thousand_separator_input_formatter.dart [hunk 9]
```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/services.dart';
+import 'package:intl/intl.dart';
+
+class ThousandsSeparatorInputFormatter extends TextInputFormatter {
+  static const separator = ',';
+  NumberFormat numberFormat = NumberFormat.decimalPattern('en');
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/helpers/thousand_separator_input_formatter.dart:6]: we don't need it anymore, so I will remove it

File: lib/view/screens/asset_receive_screen.dart [hunk 10]
```diff
@@ -75,7 +75,7 @@ class AssetReceiveScreen extends StatelessWidget with WalletUtilsMixin {
                     child: FractionallySizedBox(
                       widthFactor: 0.7,
                       child: ClipRRect(
-                        borderRadius: BorderRadiuses.radius6,
+                        borderRadius: BorderRadii.radius6,
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/screens/asset_receive_screen.dart:78]: radii is correct plural of radius word
https://www.thefreedictionary.com/radii

File: lib/view/screens/asset_send_search_screen.dart [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+              padding: Paddings.bottom32,
+              child: ListLoading(),
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  void search(String text) {
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/screens/asset_send_search_screen.dart:111]: ok

File: lib/view/screens/dapps_screen.dart [hunk 12]
```diff
@@ -5,12 +5,9 @@ class DappsScreen extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return WillPopScope(
-      onWillPop: () async {
-        return false;
-      },
-      child: const Scaffold(
-        body: SizedBox.shrink(),
+    return const Scaffold(
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/screens/dapps_screen.dart:8]: done

File: lib/view/widgets/asset_form_exception.dart [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+                        Text(
+                          state.errors(context)[formElementIndex],
+                          style: SioTextStyles.bodyS.copyWith(
+                            color: Theme.of(context).colorScheme.error,
+                          ),
+                        ),
+                      ],
+                    ),
+                  )
+                : Container();
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/widgets/asset_form_exception.dart:85]: ok, done

File: lib/view/widgets/asset_form_exception.dart [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+                            state.warnings(context)[formElementIndex],
+                            style: SioTextStyles.bodyS.copyWith(
+                              color: Theme.of(context).colorScheme.onTertiary,
+                            ),
+                          ),
+                        )
+                      ],
+                    ),
+                  )
+                : Container();
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/widgets/asset_form_exception.dart:52]: done

File: lib/view/widgets/gradient_text_button.dart [hunk 15]
```diff
@@ -1,58 +1,68 @@
...
-                          Theme.of(context).colorScheme.secondary,
-                          Theme.of(context).colorScheme.tertiary,
-                        ],
-                      ),
-                      borderRadius: BorderRadiuses.radius12,
-                    )
-                  : null,
-            ),
+          Container(
+            height: 45,
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/widgets/gradient_text_button.dart:26]: done

File: lib/data/repositories/fee_repository.dart [hunk 16]
```diff
@@ -21,7 +21,7 @@ class FeeRepository {
     );
 
     final body = res.body;
-    if (res.isSuccessful && body != null) return body.first;
+    if (res.isSuccessful && body != null && body.isNotEmpty) return body.first;
```
[the remark this answers is not in the document]
Comment (ciripel) [reply lib/data/repositories/fee_repository.dart]: Fixed.

File: lib/view/widgets/keypad_item.dart [hunk 17]
```diff
@@ -1,4 +1,5 @@
 import 'package:flutter/material.dart';
+import 'package:simplio_app/view/themes/simplio_text_styles.dart';
 
 const _keypadButtonDefaultSize = Size(100.0, 60.0);
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/widgets/keypad_item.dart:4]: ok

File: lib/data/repositories/wallet_repository.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (ciripel) [reply lib/data/repositories/wallet_repository.dart]: Fixed

File: lib/data/repositories/wallet_repository.dart (hunk 7 above)
[the remark this answers is not in the document]
Comment (ciripel) [reply lib/data/repositories/wallet_repository.dart]: I did not deleted any of this. Filip said we will need it in the future.

File: lib/view/widgets/static_progress_stepper.dart [hunk 18]
```diff
@@ -0,0 +1,220 @@
+import 'package:flutter/material.dart';
+import 'package:simplio_app/l10n/localized_build_context_extension.dart';
+import 'package:simplio_app/view/helpers/custom_painters.dart';
+import 'package:simplio_app/view/themes/simplio_text_styles.dart';
+
+const double firstStepRadius = 15.5;
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/widgets/static_progress_stepper.dart:6]: this was some remnant from the development. I will make it private for now

File: lib/view/widgets/success_check_icon.dart [hunk 19]
```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+import 'package:simplio_app/view/helpers/custom_painters.dart';
+
+class SuccessCheckIcon extends StatelessWidget {
+  final double innerRadius = 45;
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/widgets/success_check_icon.dart:5]: done. Private constant makes more sense for this case

File: pubspec.yaml (hunk 2 above)
Comment (ciripel) [reply pubspec.yaml:12]: Make a task!

File: lib/view/widgets/swipe_up_button.dart [hunk 20]
```diff
@@ -1,23 +1,22 @@
...
-  const SwipeUpButton({
+  const SwipeUpButton(
+    this.data, {
     super.key,
-    required this.child,
-    required this.height,
-    required this.swipeHeight,
-    required this.onSwipeCallback,
-    required this.onSwipeStartCallback,
+    this.height = 400,
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/view/widgets/swipe_up_button.dart:10]: ok

File: lib/view/widgets/toggle.dart (hunk 1 above)
Comment (jvrastil) [reply lib/view/widgets/toggle.dart:28]: ok, done

File: lib/data/repositories/wallet_repository.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (TomasPetrik) [reply lib/data/repositories/wallet_repository.dart]: Ok

File: lib/view/widgets/static_progress_stepper.dart (hunk 18 above)
[the remark this answers is not in the document]
Comment (TomasPetrik) [reply lib/view/widgets/static_progress_stepper.dart]: Okay.

File: pubspec.yaml (hunk 2 above)
Comment (TomasPetrik) [reply pubspec.yaml:12]: Added

File: lib/view/widgets/success_check_icon.dart (hunk 19 above)
[the remark this answers is not in the document]
Comment (TomasPetrik) [reply lib/view/widgets/success_check_icon.dart]: okay

File: lib/view/screens/asset_receive_screen.dart (hunk 10 above)
[the remark this answers is not in the document]
Comment (TomasPetrik) [reply lib/view/screens/asset_receive_screen.dart:78]: Okay. Perfect.

File: lib/view/widgets/highlighted_num_form_filed.dart [hunk 21]
```diff
@@ -0,0 +1,114 @@
...
+        FilteringTextInputFormatter.digitsOnly,
+      ],
+      focusNode: widget.focusNode,
+      style: style,
+      cursorColor: Theme.of(context).colorScheme.inverseSurface,
+      textAlignVertical: TextAlignVertical.center,
+      decoration: InputDecoration(
+          label: !widget.loading
+              ? const Padding(padding: Paddings.top10, child: Text('0'))
+              : Container(),
```
Comment (TomasPetrik) [inline lib/view/widgets/highlighted_num_form_filed.dart:74]: Use SizedBox.shrink()

File: lib/view/helpers/thousand_separator_input_formatter.dart (hunk 9 above)
[the remark this answers is not in the document]
Comment (TomasPetrik) [reply lib/view/helpers/thousand_separator_input_formatter.dart]: okay

File: lib/logic/cubit/asset_exchange_form/asset_exchange_form_cubit.dart [hunk 22]
```diff
@@ -0,0 +1,314 @@
...
+
+  void changeAmountFrom(String amountFrom) {
+    // todo: add correct price
+    const price = 19523.2;
+
+    final result = double.tryParse(amountFrom);
+    if (result != null) {
+      emit(state.copyWith(
+        amountFrom: amountFrom,
+        amountFromFiat: '${result * price}',
```
[the remark this answers is not in the document]
Comment (jvrastil) [reply lib/logic/cubit/asset_exchange_form/asset_exchange_form_cubit.dart:80]: this is only placeholder functionality (as suggests todo item on line 72). The value will be fetched from the BE

**Pull request merged**